apps/etterlatte-behandling/src/main/kotlin/behandling/BehandlingFactory.kt

```diff
@@ -257,7 +269,7 @@ class BehandlingFactory(
     }
 }
 
-data class BehandlingOgOppgave(val behandling: Behandling, val oppgave: OppgaveIntern?)
+data class BehandlingOgOppgave(val behandling: Behandling, val oppgave: OppgaveIntern?, val sendMeldingForHendelse: () -> Unit = {})
```

Comment:
Må vi sende med `sendMeldingForHendelse` som en funksjon her? Ser ikke helt verdien

Reply:
Poenget er å trekkje det ut av databasetransaksjonen som heile kallet til `opprettBehandling` er wrappa i.

Reply:
Det kan jeg være enig i men man trenger ikke å sende det som en funksjon tilbake for du trenger jo bare behandlingen sant og du vet jo behandlingstypen så du kunne jo bare gjort slik feks:


```
        val behandling =
            inTransaction {
                opprettBehandling(
                    sak.id,
                    persongalleri,
                    request.mottattDato,
                    request.kilde ?: Vedtaksloesning.GJENNY,
                    request = hentDataForOpprettBehandling(sak.id),
                ).also {
                    if (request.kilde == Vedtaksloesning.GJENOPPRETTA) {
                        oppgaveService.hentOppgaverForSak(sak.id)
                            .find { it.type == OppgaveType.GJENOPPRETTING_ALDERSOVERGANG && !it.erAvsluttet() }?.let {
                                oppgaveService.ferdigstillOppgave(it.id, brukerTokenInfo)
                            }
                    }
                } ?: throw IllegalStateException("Kunne ikke opprette behandling")
            }
                .also {
                    behandlingHendelser.sendMeldingForHendelseMedDetaljertBehandling(
                        it.behandling.toStatistikkBehandling(persongalleri),
                        BehandlingHendelseType.OPPRETTET,
                    )
                }
                .behandling
```

Reply:
Må si meg enig med @sebassonav her ☝️

Reply:
Jepp, fleire måtar å løyse det her på. Kan fint skrive om til noko sånt, sia koden brukast både frå den metoden i factory-klassa og frå routen funkar det ikkje å gjera det heilt så enkelt, men nesten, kjem oppdatering snart.

Reply:
Nei, det var ikkje så enkelt, for viss vi hamnar inne i `iverksatteEllerAttesterteBehandlinger.isNotEmpty()`-delen av if-en så skal ikkje meldinga sendast ut. Så da må vi i så fall heller sende med flagg eller noko, og da synes eg det blir vel knotete.

Reply:
Det er noe som skurrer for meg her, her prøver man å kalle opprett behandling for ting som skal bli en revurdering? 🤔 Det virker for meg som om man vet det på forhånd? Åja nei man legger ting man henter for requesten på requesten etterpå, nei det er ikke en request. Blir sjukt forvirra her nå 🤠

Reply:
Ble ikke helt ferdig med kommentaren min ser jeg :P
Poenget mitt er at det ser ut som at man vet på forhånd om man skal opprette revurdering eller ikke så man kunne flyttet den logikken et hakk ut og faktisk kun ha "opprett behandling"

Reply:
Ja, den biten er skikkeleg forvirrande. Hadde vore deilig å splitte det heilt opp.